examples/memcached.md

```diff
@@ -0,0 +1,369 @@
+
+It is not necessary to check out the PMEM-CSI source code.
+
+## PMEM as DRAM replacement
+
+Memcached can map PMEM into its address space and then store the bulk
+of the cache there. The advantage is higher capacity and (depending on
+pricing) less costs.
+
+The cache could be stored in a persistent volume and reused after a
+restart (see next section), but when that isn't the goal, a
+[deployment with one CSI ephemeral inline
+volume](/deploy/kustomize/memcached/ephemeral/memcached-ephemeral.yaml)
+per memcached pod is very simple:
+- A Kubernetes deployment manages the memcached instance(s).
+- In the pod template, one additional PMEM volume of a certain size is
+  requested.
+- An init container prepares that volume for use by memcached:
+  - It determines how large the actual file can be that memcached will use.
+  - It changes volume ownership so that memcached can run as non-root
+    process.
+- A wrapper script adds the `--memory-file` and `--memory-limit` parameters
+  when invoking memcached, which enables the use of PMEM.
+
+In this mode, the PMEM-CSI driver is referenced through its name
+(usually `pmem-csi.intel.com`). No storage classes are needed. That
+name and other parameters in the deployment can be modified with
+[`kustomize`](https://github.com/kubernetes-sigs/kustomize). Here's
+how one can change the namespace, volume size or add additional
+command line parameters:
+
+```sh
+mkdir -p my-memcached-deployment
+
+cat >my-memcached-deployment/kustomization.yaml <<EOF
+namespace: demo
+bases:
+ - github.com/intel/pmem-csi/deploy/kustomize/memcached/ephemeral
+patchesStrategicMerge:
+- update-deployment.yaml
+EOF
+
+cat >my-memcached-deployment/update-deployment.yaml <<EOF
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: pmem-memcached
+spec:
+  template:
+    spec:
+      containers:
+      - name: memcached
+        env:
+        - name: MEMCACHED_ARGS
+          value: --conn-limit=500
+      volumes:
+      - name: data-volume
+        csi:
+          volumeAttributes:
+            size: 500Mi
+EOF
+```
+
+These commands then create the namespace and the deployment:
+```sh
+kubectl create ns demo
+kubectl create --kustomize my-memcached-deployment
+```
+
+The expected output is:
+```
+namespace/demo created
+service/pmem-memcached created
+deployment.apps/pmem-memcached created
+```
+
+Eventually, there will be one pod with memcached running:
+```sh
+kubectl wait --for=condition=Ready -n demo pods -l app.kubernetes.io/name=pmem-memcached
+```
+
+```
+pod/pmem-memcached-96f6b5986-hdx6n condition met
+```
+
+How to access a service from remote varies between cluster
+installations and how they handle incoming connections. Therefore we
+use [`kubectl
+port-forward`](https://kubernetes.io/docs/tasks/access-application-cluster/port-forward-access-application-cluster/)
+to access the service. Run this command in one shell and keep it
+running there:
+```sh
+kubectl port-forward -n demo service/pmem-memcached 11211
```

Comment:
I think combining the output with the command makes sense in this case if you prepend the command with $ so users know it is a command. It is the telnet session where you don't have the $ anymore where I think separating and indicating one is output makes sense. 

```
$ kubectl port-forward -n demo service/pmem-memcached 11211
Forwarding from 127.0.0.1:11211 -> 11211
Forwarding from [::1]:11211 -> 11211
```